lld/test/MachO/section-order.s

```diff
@@ -1,4 +1,3 @@
-# REQUIRES: x86
```

Comment:
Probably not intended?

Reply:
It is. I wanted to make sure it ran since I knew this change affected it, and since it works fine I figured we should remove it. I guess this is sort of by accident since it's just directives with no instructions?

Reply:
We're still passing an x86 triple to `mc`. Or is that fine?

Reply:
I got turned around re: how "Requires" works. Reverted, thanks!